TESTS/mbed_hal/qspi/qspi_test_utils.cpp

```diff
@@ -0,0 +1,283 @@
+qspi_status_t write_enable(Qspi &qspi)
+{
+    uint8_t reg[QSPI_STATUS_REGISTER_SIZE];
+    qspi.cmd.build(QSPI_CMD_WREN);
+
+    if (qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    memset(reg, 0, QSPI_STATUS_REGISTER_SIZE);
+    if (read_register(STATUS_REG, reg, QSPI_STATUS_REGISTER_SIZE, qspi) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+
+    return ((reg[0] & STATUS_BIT_WEL) != 0 ? QSPI_STATUS_OK : QSPI_STATUS_ERROR);
+}
+
+qspi_status_t write_disable(Qspi &qspi)
+{
+    uint8_t reg[QSPI_STATUS_REGISTER_SIZE];
+    qspi.cmd.build(QSPI_CMD_WRDI);
+
+    if (qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    memset(reg, 0, QSPI_STATUS_REGISTER_SIZE);
+    if (read_register(STATUS_REG, reg, QSPI_STATUS_REGISTER_SIZE, qspi) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+
+    return ((reg[0] & STATUS_BIT_WEL) == 0 ? QSPI_STATUS_OK : QSPI_STATUS_ERROR);
+}
+
+WEAK void log_security_register(Qspi &qspi)
+{
+    qspi_status_t ret;
+    uint8_t reg[QSPI_SECURITY_REGISTER_SIZE];
+
+    ret = read_register(SECURITY_REG, reg, QSPI_SECURITY_REGISTER_SIZE, qspi);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    for (int j = 0; j < QSPI_SECURITY_REGISTER_SIZE; j++) {
+        printf("security register %d: ", j);
+        for(int i = 0; i < 8; i++) {
+            printf("%s ", ((reg[j] & (1 << i)) & 0xFF) == 0 ? "0" : "1");
+        }
+        printf("\r\n");
+    }
+}
+
+WEAK void log_status_register(Qspi &qspi)
+{
+    qspi_status_t ret;
+    uint8_t reg[QSPI_STATUS_REGISTER_SIZE];
+
+    ret = read_register(STATUS_REG, reg, QSPI_STATUS_REGISTER_SIZE, qspi);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    for (int j = 0; j < QSPI_STATUS_REGISTER_SIZE; j++) {
+        printf("status register %d: ", j);
+        for(int i = 0; i < 8; i++) {
+            printf("%s ", ((reg[j] & (1 << i)) & 0xFF) == 0 ? "0" : "1");
+        }
+        printf("\r\n");
+    }
+}
+
+WEAK void log_config_register(Qspi &qspi)
+{
+    qspi_status_t ret;
+    uint8_t reg[QSPI_CONFIGURATION_REGISTER_SIZE];
+
+    ret = read_register(CONFIG_REG, reg, QSPI_CONFIGURATION_REGISTER_SIZE, qspi);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    for (int j = 0; j < QSPI_CONFIGURATION_REGISTER_SIZE; j++) {
+        printf("config register %d: ", j);
+        for(int i = 0; i < 8; i++) {
+            printf("%s ", ((reg[j] & (1 << i)) & 0xFF) == 0 ? "0" : "1");
+        }
+        printf("\r\n");
+    }
+}
+
+WEAK qspi_status_t dual_enable(Qspi &qspi)
+{
+    return QSPI_STATUS_OK;
+}
+
+WEAK qspi_status_t dual_disable(Qspi &qspi)
+{
+    return QSPI_STATUS_OK;
+}
+
+WEAK qspi_status_t quad_enable(Qspi &qspi)
+{
+    uint8_t reg_data[QSPI_STATUS_REGISTER_SIZE];
+
+    reg_data[0] = STATUS_BIT_QE;
```

Comment:
SFDP standard defines Quad Enable procedure depending on QER  bits value.
This procedure of matches QER setup 010b where QE is bit 6 of status register.
Do we require a more general approach that handles Quad Enable for all QER value scenarios?

Reply:
Good point, could be done in next step.
But not sure if we really need such flexibility as SFDP provides in test layer.
The idea of QSPI tests was to check basic features, common for all targets/memory chips,
just to be sure that HAL API implementations works
@0xc0170 @SenRamakri

Reply:
@offirko This is utils functions for the driver test that has no SFDP knowledge. SFDP will be part of the device block driver (tested there). What would you suggest here?